src/app/dso-shared/dso-edit-metadata/dso-edit-metadata-value-field/dso-edit-metadata-value-field-loader/dso-edit-metadata-value-field.decorator.ts

```diff
@@ -0,0 +1,59 @@
+import { Context } from '../../../../core/shared/context.model';
+import { hasValue } from '../../../../shared/empty.util';
+import {
+  DEFAULT_CONTEXT,
+  DEFAULT_THEME,
+  resolveTheme,
+} from '../../../../shared/object-collection/shared/listable-object/listable-object.decorator';
+import { DsoEditMetadataAuthorityFieldComponent } from '../dso-edit-metadata-authority-field/dso-edit-metadata-authority-field.component';
+import { DsoEditMetadataEntityFieldComponent } from '../dso-edit-metadata-entity-field/dso-edit-metadata-entity-field.component';
+import { EditMetadataValueFieldType } from '../dso-edit-metadata-field-type.enum';
+import { DsoEditMetadataTextFieldComponent } from '../dso-edit-metadata-text-field/dso-edit-metadata-text-field.component';
+
+export type MetadataValueFieldComponent =
+  typeof DsoEditMetadataTextFieldComponent |
+  typeof DsoEditMetadataEntityFieldComponent |
+  typeof DsoEditMetadataAuthorityFieldComponent;
+
+export const map = new Map<EditMetadataValueFieldType, Map<Context, Map<string, MetadataValueFieldComponent>>>([
+  [EditMetadataValueFieldType.PLAIN_TEXT, new Map([
+    [DEFAULT_CONTEXT, new Map([[DEFAULT_THEME, DsoEditMetadataTextFieldComponent]])],
+  ])],
+  [EditMetadataValueFieldType.ENTITY_TYPE, new Map([
+    [DEFAULT_CONTEXT, new Map([[DEFAULT_THEME, DsoEditMetadataEntityFieldComponent]])],
+  ])],
+  [EditMetadataValueFieldType.AUTHORITY, new Map([
+    [DEFAULT_CONTEXT, new Map([[DEFAULT_THEME, DsoEditMetadataAuthorityFieldComponent]])],
+  ])],
+]);
+
+export const DEFAULT_EDIT_METADATA_FIELD_TYPE = EditMetadataValueFieldType.PLAIN_TEXT;
+
+/**
+ * Getter to retrieve a matching component by entity type, metadata representation and context
+ *
+ * @param type The edit metadata field type
+ * @param context The context to match
+ * @param theme the theme to match
+ */
+export function getDsoEditMetadataValueFieldComponent(type: EditMetadataValueFieldType, context: Context = DEFAULT_CONTEXT, theme = DEFAULT_THEME) {
```

Comment:
Would it be possible to add basic specs for this function?  Codecov is warning that every line of this function appears to not be covered by tests.   I do realize though that this function is not too complex, but it might be nice to have tests nonetheless.  (But if it's too difficult / time consuming, we can skip it)

Reply:
I started creating some custom tests for this, but I don't think we should include those, because if people want to customize some components and extend the `map` with their own components, the tests would also fail. To prevent this we would need to override the map in the tests, but this would require us from switching that variable from a `const` to a `let`, and I'm not sure that would be a good idea

Reply:
I see your point.  We can keep this as-is then.  As I noted initially, it was more of a "nice to have" thing, if it didn't cause problems.